backend/src/main/java/pl/edu/agh/awiteks_backend/api/UserController.java

```diff
@@ -31,6 +32,24 @@ public Optional get(@PathVariable int id) {
         return super.get(id);
     }
 
+    @GetMapping(value = "/users/{id}/plants", produces = "application/json")
+    public List<Plant> getPlants(@PathVariable int id) {
+        var user = super.get(id);
+        if(user.isPresent())
+            return user.get().getUserPlants();
+        else
+            return new ArrayList<Plant>();
+    }
+
+    @GetMapping(value = "/users/{id}/favPlants", produces = "application/json")
+    public List<Plant> getFavPlants(@PathVariable int id) {
+        var user = super.get(id);
+        if(user.isPresent())
+            return user.get().getUserPlants().stream().filter(Plant::isFavourite).toList();
+        else
+            return new ArrayList<Plant>();
+    }
+
```

Comment:
W jaki sposób będzie można dodawać/usuwać roślinę do/z ulubionych przez API? Bo tego nie widzę. Przez `PlantControlers` przez updatetowanie `PUT`? Można dodać najwyżej osobny endpoint `PUT` w `PlantControler`, który zmienia tą wartość

Reply:
Mamy już metodę update pod PUT, która prawdopodobnie potrafi to zrobić. Chcemy coś bardziej wyspecjalizowanego czy wystarczy?